### DISCUSSION TITLE
How to perform message passing in HeteroGraph with empty node,edge types

### DISCUSSION THREAD

**Discussion by DogitoErgoSum** [pyg-team/pytorch_geometric#4266] [Q&A] [answered]
Hi,
My main goal is to perform HeteroGraph classification. Not all graphs will have all node types and will have different node type combinations, as shown below:

![image](https://user-images.githubusercontent.com/51344466/158429985-cabb3344-1112-4a47-a556-29c37da5090a.png)
Here  L0_S5_F114 has no nodes present.
I tried to perform a simple message passing on a graph like with the following code but it failed.

`conv = HeteroConv({edge_type: GraphConv((-1, -1), 5) for edge_type in data.edge_types}, aggr='sum').to(device)`

`conv(data.x_dict, data.edge_index_dict, edge_weight_dict=data.edge_attr_dict)`

![image](https://user-images.githubusercontent.com/51344466/158431031-1c0222f9-b88d-4cb5-b235-476f16987335.png)

What am I doing wrong or what should I be doing?

**Comment by DogitoErgoSum** [comment]
The problem here is that there is message passing occurring in empty node types, but I need to keep the empty node types to be able to create batches.
How do I perform message passing only on non-empty nodes?

**Comment by rusty1s** [comment, marked as answer]
Thanks for reporting. I will try to reproduce this on my end. If you also have a script to reproduce, please let me know. Message passing with empty node types should definitely work in PyG >= 2.0.4

**Reply by DogitoErgoSum** [reply]
![image](https://user-images.githubusercontent.com/51344466/158444453-e4355d9d-4aa3-407d-955d-6754e7a22725.png)
The version seems to be the problem, i will check if it fixed.
TY

**Reply by rusty1s** [reply]
You can simply try to install via `pip install torch-geometric==2.0.4`. Let me know.

**Reply by DogitoErgoSum** [reply]
It works now, ty for your fast and accurate response
